Tasks/S3Upload/TaskOperations.ts

```diff
@@ -135,14 +135,22 @@ export class TaskOperations {
                             break
 
                         case awsKeyManagementValue: {
-                            request.ServerSideEncryption = this.taskParameters.encryptionAlgorithm
-                            request.SSEKMSKeyId = this.taskParameters.kmsMasterKeyId
+                            if (this.taskParameters.encryptionAlgorithm) {
+                                request.ServerSideEncryption = this.taskParameters.encryptionAlgorithm
+                            }
+                            if (this.taskParameters.kmsMasterKeyId) {
+                                request.SSEKMSKeyId = this.taskParameters.kmsMasterKeyId
+                            }
                             break
                         }
 
                         case customerKeyManagementValue: {
-                            request.SSECustomerAlgorithm = this.taskParameters.encryptionAlgorithm
-                            request.SSECustomerKey = this.taskParameters.customerKey
+                            if (this.taskParameters.encryptionAlgorithm) {
+                                request.SSECustomerAlgorithm = this.taskParameters.encryptionAlgorithm
+                            }
+                            if (this.taskParameters.customerKey !== Buffer.from([])) {
```

Comment:
Is there another way to verify? For example, if we can't compare against something like length, could we create a static readonly instance of the initialized value used in TaskParameters.ts (line 38) and compare against that.

Reply:
same, changed to length